tests/test_file_read_backwards.py

```diff
@@ -161,26 +160,49 @@ def test_file_with_new_lines_and_some_accented_characters_in_chunk_size(self):
                     expected_lines,
                     lines_read,
                     msg="Test with {0} of new line {1!r} followed by {2} of \\xc3\\xa9".format(number_of_new_lines,
-                                                                                          new_line, chunk_size))
+                                                                                               new_line, chunk_size))
 
     def test_unsupported_encoding(self):
         """Test when users pass in unsupported encoding, NotImplementedError should be thrown."""
         with self.assertRaises(NotImplementedError):
             _ = FileReadBackwards(self.empty_file.name, encoding="not-supported-encoding")  # noqa: F841
 
-    def test_reusing_iterator(self):
-        f = FileReadBackwards(self.long_file.name, encoding='utf-8')
-        lines_read = deque()
-        for l in f:
-            lines_read.appendleft(l)
-        lines_read2 = deque()
-        for l2 in f:
-            lines_read2.appendleft(l2)
-        self.assertEqual(
-            lines_read,
-            lines_read2,
-            msg="Test that verifies that it creates a new iterator everytime"
-        )
+    def test_file_with_one_line_of_text_readline(self):
+        """Test a file with a single line of text followed by a new line."""
+        s = "Line0"
+        for new_line in new_lines:
+            temp_file = helper_create_temp_file((l for l in [s, new_line]))
+            with FileReadBackwards(temp_file.name) as fp:
+                l = fp.readline()
+                expected_line = s + os.linesep
+                self.assertEqual(l, expected_line)
+
+                # the file contains only 1 line
+                second_line = fp.readline()
+                expected_second_line = ""
+                self.assertEqual(second_line, expected_second_line)
+
+    def test_file_with_two_lines_of_text_readline(self):
+        """Test a file with a two lines of text followed by a new line."""
+        line0 = "Line0"
+        line1 = "Line1"
+        for new_line in new_lines:
```

Comment:
Maybe we can start getting benefits by switching over `pytest` and use `@pytest.mark.parametrize(['newline'], new_line)`?
And most tests will be merged into one, as it's just the input and expected results that will change.